File: js/src/1812-squareIsWhite.js
/*
 * @lc app=leetcode.cn id=1812 lang=javascript
 * [1812] 判断国际象棋棋盘中一个格子的颜色
 */

// 难度：简单
// 考点：数组基本操作

/**
 * @param {string} coordinates
 * @return {boolean}
 */
// 80 ms, 在所有 JavaScript 提交中击败了80.62%
const squareIsWhite = function(coordinates) {
  const first = coordinates[0];
  const second = parseInt(coordinates[1]);
  if (first === 'a' || first === 'c' || first === 'e' || first === 'g') {
    return second % 2 === 0;
  } else {
    return second % 2 !== 0;
  }
};

export { squareIsWhite };

File: python/src/1812-squareIsWhite.py
# https://leetcode.cn/problems/determine-color-of-a-chessboard-square/
class Solution:
    def squareIsWhite(self, coordinates: str) -> bool:
        first = coordinates[0]
        second = int(coordinates[1], 10)
        if (first == 'a' or first == 'c' or first == 'e' or first == 'g'):
            if second % 2 == 0:
                return True
            else:
                return False
        else:
            if second % 2 == 0:
                return False
            else:
                return True
